csaf_2_1/mandatoryTests/mandatoryTest_6_1_36.js

```diff
@@ -0,0 +1,221 @@
+
+const validate = ajv.compile(inputSchema)
+
+/**
+ * This implements the mandatory test 6.1.36 of the CSAF 2.1 standard.
+ *
+ * @param {any} doc
+ */
+export function mandatoryTest_6_1_36(doc) {
+  /*
+    The `ctx` variable holds the state that is accumulated during the test ran and is
+    finally returned by the function.
+   */
+  const ctx = {
+    /** @type {Array<{ instancePath: string; message: string }>} */
+    errors: [],
+    isValid: true,
+  }
+
+  if (!validate(doc)) {
+    return ctx
+  }
+
+  for (const [
+    vulnerabilityIndex,
+    vulnerability,
+  ] of doc.vulnerabilities.entries()) {
+    vulnerability.remediations?.forEach((remediation, remediationIndex) => {
+      const category = remediation.category
+      if (!category) return
+
+      /**
+       * This map holds the discovered product ids for the remediation and maps them to
+       * the set of corresponding product status names. Later we can check this map to
+       * find out if there are any contradicting remediations.
+       *
+       * @type {Map<string, Set<string>>}
+       */
+      const productToProductStatusNamesMap = new Map()
+
+      /**
+       * This function adds all product status names for the given product id to the
+       * `productMap`. If the product does not yet exist in the map, it is added.
+       *
+       * @param {string} id
+       */
+      const collectProductStatusNames = (id) => {
+        const productStatusNames =
+          /*
+            To speed things up we first check if the product status names where already
+            collected and do not search again. The product names are always for a
+            product in the same vulnerability.
+           */
+          productToProductStatusNamesMap.get(id) ??
+          new Set(
+            /** @type {string[]} */ (
+              Object.entries(vulnerability.product_status ?? {})
+                .filter((e) =>
+                  Array.isArray(e[1]) ? e[1].includes(id) : false
+                )
+                .map((e) => e[0])
+            )
+          )
+        productToProductStatusNamesMap.set(id, productStatusNames)
+      }
+
+      remediation.product_ids?.forEach(collectProductStatusNames)
+
+      remediation.group_ids?.forEach((id) => {
+        const group = doc.product_tree?.product_groups?.find(
+          (g) => g.group_id === id
+        )
+        if (!group) return
+        group.product_ids?.forEach(collectProductStatusNames)
+      })
+
+      for (const [
+        productId,
+        productStatusNames,
+      ] of productToProductStatusNamesMap) {
+        for (const productStatus of productStatusNames) {
+          if (prohibitionRuleMap.get(category)?.has(productStatus)) {
+            ctx.errors.push({
+              instancePath: `/vulnerabilities/${vulnerabilityIndex}/remediations/${remediationIndex}`,
+              message: `contradicting remediation product status combination for product id "${productId}": ${category}, ${productStatus}`,
```

Comment:
Maybe, we can improve the error message?
```suggestion
              message: `contradicting remediation product status combination for product id "${productId}": product status is ${productStatus} but remediation category is ${category}`,

Reply:
@domachine I guess, I'm missing a response to this comment...

Reply:
What do you think about the suggested error message in https://github.com/secvisogram/csaf-validator-lib/pull/257/files#r2074269857? Please ensure that both follow the same pattern.